labview/labview_types.h

```diff
@@ -3,6 +3,26 @@
 #ifndef LABVIEW_TYPES_H
 #define LABVIEW_TYPES_H
 
+/**
+ * Structure that represents LabVIEW Cluster (items: string key, string value, int32 dataType)
+ */
+typedef struct {
+	LStrHandle key;
+	LStrHandle value;
+    int32_t dataType;
+} KeyValuePairRec;
+/**
+ * Structure that represents LabVIEW Cluster Array
+*/
+typedef struct {
+	int32_t dimSize;
+	KeyValuePairRec elt[1];
+} KeyValuePairArr;
+/**
+ * Handle to LabVIEW Cluster Array, used to pass it between LabVIEW and C
+*/
+typedef KeyValuePairArr **KeyValuePairArrHdl;
```

Comment:
I don't get this typedef

Reply:
is a handle (pointer to pointer). Handle is send by LabVIEW if we use "adapt to type" for Call library node.

Reply:
Labview aloocates the memory for teh structure -> First byte of this structure is a pointer. To make a memory management easy - in case of structure must be realocated (resized etc.) LabVIEW keeps a pointer to a pointer -> handle.